spec/controllers/order_controller_spec.rb

```diff
@@ -0,0 +1,211 @@
+require "rails_helper"
+include SessionsHelper
+
+RSpec.shared_examples "user don't log in" do
+  before {get :new}
+  it "is store forwarding_url" do
+    expect(session[:forwarding_url]).not_to be_nil
+  end
+
+  it "redirect to login path" do
+    expect(response).to redirect_to new_session_path
+  end
+end
+
+RSpec.describe OrdersController, type: :controller do
+  let!(:user) {FactoryBot.create :user}
+  let!(:order) {FactoryBot.create(:order, user_id: user.id)}
+  let!(:product) {FactoryBot.create :product}
+
+  describe "GET #new" do
+    context "success when valid attributes" do
+      context "when curren cart empty" do
+        before do
+          log_in user
+          get :new
+        end
+
+        it "assign @order" do
+          expect(assigns(:order)).not_to be_nil
+        end
+
+        it "assign @cart_items " do
+          expect(assigns(:cart_items).empty?).to be true
+        end
+      end
+
+      context "when curren cart not empty" do
+        before do
+          session[:cart] = [{product_id: product.id, quantity: 1}]
+          log_in user
+          get :new
+        end
+
+        it "assign @cart_items " do
+          expect(assigns(:cart_items).empty?).to be false
+        end
+      end
+
+      it_behaves_like "user don't log in"
+    end
+  end
+
+  describe "POST #create" do
+    let(:order_params) {FactoryBot.build(:order).as_json}
+
+    context "success when valid attributes" do
+      before do
+        session[:cart] = [{product_id: product.id, quantity: 1}]
+        log_in user
+        get :create, params: {
+          order: order_params
+        }
+      end
+
+      it "assign @orders " do
+        expect(assigns(:order)).not_to be_nil
+      end
+
+      it "reduce product quantity after create order success" do
+        expect(Product.all.second.quantity).to eq(product.quantity - 1)
```

Comment:
`Product.all.second` thành `Product.second` cần j `.all` thừa á
cơ mà chỗ này ko ổn, sao em bjk nó là second =))
nên find rồi đi so sánh nhé
`expect(Product.find_by(id: product.id)&.quantity).to eq(product.quantity - 1)`